### ISSUE TITLE
General solution for waiting REST service

### ISSUE THREAD

**Issue by ricoapon** [GreenT13/TaalmaatjesWebsite#3]
When doing anything that requires a reaction from the REST service, you can still click around and do anything in Angular. Also functions complete and return something while the subscribe method has not yet finished. This created some wonky stuff.

There are two things to find out:
1. Function returning a final answer instead of a promise (and really waiting for it).
2. Some sort of overlay with spinning circle, indication that the website is loading something.

**Comment by ricoapon** [comment]
Fixed overlay in 390ab227a6f0bd9c23eb176a7f5766978fb44a5d, which is issue 2.

I guess issue 1 does not need to be solved if I wait on every promise/observable. Thus I close the issue. If I still need to figure that out (which I probably will, but not for this case), I will make a new ticket.
